src/main/scala/clump/ClumpSource.scala

```diff
@@ -5,11 +5,15 @@ import scala.collection.generic.CanBuildFrom
 
 class ClumpSource[T, U] private (val fetch: Set[T] => Future[Map[T, U]], val maxBatchSize: Int = Int.MaxValue) {
 
-  def get(inputs: T*): Clump[List[U]] = get(inputs.toList)
+  def list(inputs: T*): Clump[List[U]] = list(inputs.toList)
 
-  def get(inputs: List[T]): Clump[List[U]] = Clump.collect(inputs.map(get))
+  def list(inputs: List[T]): Clump[List[U]] = Clump.collect(inputs.map(apply))
 
-  def get(input: T): Clump[U] = new ClumpFetch(input, ClumpContext().fetcherFor(this))
+  def get(input: T): Clump[Option[U]] = apply(input).optional
+
+  def getOrElse(input: T,  default: => U): Clump[U] = apply(input).orElse(Clump.value(default))
+
+  def apply(input: T): Clump[U] = new ClumpFetch(input, ClumpContext().fetcherFor(this))
```

Comment:
This method doesn't have a behavior similar to ```Map.apply```, that throws an exception if the element isn't found. This implementation returns values for any provided input.